sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_responses.py

```diff
@@ -0,0 +1,38 @@
+# The MIT License (MIT)
+# Copyright (c) 2024 Microsoft Corporation
+
+from azure.core.utils import CaseInsensitiveDict
+
+
+class CosmosDictResponse(dict):
+    def __init__(self, original_dict, response_headers):
+        if original_dict is None:
+            original_dict = {}
+        super().__init__(original_dict)
+        self._response_headers = response_headers
+
+    @property
+    def response_headers(self) -> CaseInsensitiveDict:
+        """Returns the response headers associated to this result
+
+        :return: Dict of response headers
+        :rtype: ~azure.core.CaseInsensitiveDict
+        """
+        return self._response_headers
+
+
+class CosmosListResponse(list):
+    def __init__(self, original_list, response_headers):
```

Comment:
Same here:
```suggestion
    def __init__(self, original_list: Iterable[Any], /, *, response_headers: Mapping[str, Any) -> None:
```
I'm not entirely sure if using `Iterable[Any]` here will work with the other type hints we have for the APIs that return lists - it might need to be Iterable[Dict[str, Any]] or something, otherwise customer might get a useless return type of `Any`....

Reply:
this is all meant to be used internally though, a user should not be creating their own response items - not entirely sure the typehint would be relevant to them so we should be able to make it whatever makes sense to us